### PR TITLE
Refactor '/query-airflow-database' doc to use AIRFLOW_CONN_AIRFLOW_DB (for #1130)

### PR DESCRIPTION
**Pull request by paolaperaza**
Addresses [#1130](https://github.com/astronomer/issues/issues/1130)

@schnie @kaxil Can you two take a look at this? I essentially took Kaxil's comment in the issue (https://github.com/astronomer/issues/issues/1130#issuecomment-6254457810) and replaced our current instructions with those guidelines + example DAG. Tried to be as clear as possible that:
- The `airflow_db` connection will be incorrectly populated in the Airflow UI (but will still work)
- Users do not have to set the `AIRFLOW_CONN_AIRFLOW_DB` Env Var themselves

I also added:
- Risks associated with accessing the DB
- Use cases for accessing the DB
- A note on how to pull those postgres creds for Enterprise customers (not required obviously but we do have a doc on it - https://www.astronomer.io/docs/ee-administration-postgres-creds/)

If you two approve, would love to have someone from support QA test this one more time. Thank ya!

### CODE REVIEW

File: v0.13/query-airflow-database.md [hunk 1]
```diff
@@ -8,33 +8,49 @@ slug: "query-airflow-database"
...
+> Note: Airflow's "Ad-Hoc Query" feature used to be a common way to test DB connections and query the Airflow Metadata Database via the Airflow UI but was recently deprecated in 1.10 for security reasons.
 
-When developing locally, you'll need to set up your local Postgres connection.
+### Risk Associated with Database Access
 
-**1. Add `libstdc++` to `packages.txt`**
+As noted above, your Airflow Deployment's Metadata Database on Astronomer is either hosted by your team if you're using Astronomer Enterprise or by ours if you're running on Astronomer Cloud. In both cases, it's worth noting the risks associated with accessing Airflow's Database given its importance to the Scheduler's performance.
 
-Airflow uses Pandas to power the Ad Hoc Query engine, which requires this additional dependency.
+We _strongly_ recommend users do not write to the database directly as it can compromise both the integrity of your Airflow Deployment and our team's ability to support you in the case of an issue.
```
Comment (schnie) [inline v0.13/query-airflow-database.md:28]: We should follow this up with the addition of a _read only_ user that can be populated here for customers to use. Good advice for now.

File: v0.13/query-airflow-database.md [hunk 2]
```diff
@@ -54,46 +70,102 @@ Your connection should look something like:
...
+Port: 6543
+```
 
-- Navigate to `Data Profiling` > `Ad Hoc Query` on the Airflow UI
-- Select `airflow_db` as the database from the menu
-- Run a sample query
+### Populating the `airflow_db` Connection on Astronomer Enterprise
 
-![Ad Hoc Query Page](https://assets2.astronomer.io/main/docs/query-postgres/query-postgres-adhocquery.png)
+On Astronomer, your deployment's Postgres credentials are also stored as a Kubernetes Secret in your deployment's namespace. For users with `kubectl` access to the cluster who want to properly populate the `airflow_db` connection in the Airflow UI with the proper credentials, follow our nstructions [here](https://www.astronomer.io/docs/ee-administration-postgres-creds/).
```
Comment (schnie) [inline v0.13/query-airflow-database.md:166]:
```suggestion
On Astronomer, your deployment's Postgres credentials are also stored as a Kubernetes Secret in your deployment's namespace. For users with `kubectl` access to the cluster who want to properly populate the `airflow_db` connection in the Airflow UI with the proper credentials, follow our instructions [here](https://www.astronomer.io/docs/ee-administration-postgres-creds/).
```

Comment (paolaperaza) [reply v0.13/query-airflow-database.md]: ah, thank u for the catch @schnie !

File: v0.13/query-airflow-database.md [hunk 3]
```diff
@@ -8,33 +8,49 @@ slug: "query-airflow-database"
 
 ## Overview
 
-At Astronomer, each Airflow deployment is eqiupped with a PostgreSQL database that serves as Airflow's underlying metadata database and your Airflow scheduler's source of truth.
+On Astronomer, each Airflow deployment is eqiupped with a PostgreSQL database that serves as Airflow's underlying metadata database and your Airflow scheduler's source of truth.
```
Comment (kaxil) [inline v0.13/query-airflow-database.md:11]:
```suggestion
On Astronomer, each Airflow deployment is equipped with a PostgreSQL database that serves as Airflow's underlying metadata database and your Airflow scheduler's source of truth.
```

File: v0.13/query-airflow-database.md [hunk 4]
```diff
@@ -54,47 +70,103 @@ Your connection should look something like:
...
-port: 6543
+Conn Id: `airflow_db
+Conn Type: MySQL
+Host: mysql
+Schema: airflow
+Login: root
 ```
 
-For example, if your deployment name was `quasaric-sun-9051`, the connection information would be the following:
+While this information is incorrect, the underlying connection will still succeed. In upcoming versions of Astronomer, we intend to make sure this connection is properly populated.
```
Comment (kaxil) [inline v0.13/query-airflow-database.md:97]: Even if they do edit this connection in the UI (which is stored in metadata DB), `AIRFLOW_CONN_AIRFLOW_DB` will have higher preference. 

Airflow's Precedence for searching a connection is following:

1. Secret's Backend (From Airflow 1.10.10 and if the Secret's Backend is defined)
2. Environment Variable
3. Metadata DB

Comment (paolaperaza) [reply v0.13/query-airflow-database.md:97]: Oh interesting - that's helpful, thanks @kaxil I'll adjust wording a bit.

Comment (paolaperaza) [reply v0.13/query-airflow-database.md]: Adjusted to -

> While this information is incorrect in the Airflow UI, the underlying connection will still succeed, as connections set by an Environment Variable take precendence over connection details in the Astronomer UI and in the Metadata Database itself.

**Pull request merged**